go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -264,21 +265,21 @@ func newOvnAddressSets(nbClient libovsdbclient.Client, name string, ips []net.IP
 
 	ip4ASName, ip6ASName := MakeAddressSetName(name)
 	if config.IPv4Mode {
-		v4set, err = newOvnAddressSet(nbClient, ip4ASName, v4IPs)
+		v4set, err = newOvnAddressSet(nbClient, ip4ASName, v4IPs, false)
 		if err != nil {
 			return nil, err
 		}
 	}
 	if config.IPv6Mode {
-		v6set, err = newOvnAddressSet(nbClient, ip6ASName, v6IPs)
+		v6set, err = newOvnAddressSet(nbClient, ip6ASName, v6IPs, false)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return &ovnAddressSets{nbClient: nbClient, name: name, ipv4: v4set, ipv6: v6set}, nil
 }
 
-func newOvnAddressSet(nbClient libovsdbclient.Client, name string, ips []net.IP) (*ovnAddressSet, error) {
+func newOvnAddressSet(nbClient libovsdbclient.Client, name string, ips []net.IP, inRecursion bool) (*ovnAddressSet, error) {
```

Comment:
so IIRC, L267 and L273 are the only places we call `newOvnAddressSet` from right? Kinda makes the whole `inRecursion` variable moot at this point since its always false. Do you think we should maybe just go with the error comparison part for the retry?

I know the aim is to avoid infinite recursion issues, but if let's say constraint violation was happening because we have 2 duplicated entries in NBDB, would we get stuck on this infinite loop? I think yes, because this bool wouldn't protect us, we might need a better way to cut off the loop.

Reply:
@tssurya so lines 267 and 274 are the non recursive calls to newOvnAddressSet.

There is not need to recurse more than once, because the second call will follow the update codepath in that logic and no longere return a constrain violation.
So, the only one time we 'start over' is in line 304. I do not see an infinite loop here, but I may be
not accounting for something?!? There may be other ways of accomplishing this, so we should change to the way that is clearest to understand.

Reply:
ah nvm, sorry I missed that part out, right makes sense that you call it with true from when you recurse.